In [19]:
import pandas as pd

df = pd.read_csv("dataset_dialogos.csv")
df.head(), df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 155507 entries, 0 to 155506
Data columns (total 4 columns):
 #   Column      Non-Null Count   Dtype 
---  ------      --------------   ----- 
 0   Unnamed: 0  155507 non-null  int64 
 1   speaker     155507 non-null  object
 2   dialog      155507 non-null  object
 3   length      155507 non-null  int64 
dtypes: int64(2), object(2)
memory usage: 4.7+ MB


(   Unnamed: 0 speaker                                             dialog  \
 0           0    amlo    Amigas, amigos, paisanas, paisanos de Palenque:   
 1           1    amlo  Me da mucho gusto estar de nuevo en trabajos d...   
 2           2    amlo  Antes de entrar en materia, quiero enviar un s...   
 3           3    amlo  Di la instrucción de que se mantengan trabajan...   
 4           4    amlo  Quiero decirles a mis paisanos que padecen, qu...   
 
    length  
 0      47  
 1      91  
 2     471  
 3     338  
 4     113  ,
 None)

#### Tiene 155,507 registros y 4 columnas: Unnamed: 0 , donde es un índice redundante por lo cual es mejor eliminar la columna Unnamed: 0, ya que parece un índice innecesario, despues voy a darle una revisada a los valores únicos en speaker para verificar si hay inconsistencias en los nombres y tambien buscar valores nulos o duplicados en dialog.

In [21]:
# Revisar valores únicos en 'speaker'
unique_speakers = df['speaker'].unique

# Contar diálogos duplicados
duplicated_dialogs = df.duplicated(subset=['dialog']).sum()

# Estadísticas de la longitud del diálogo
length_stats = df['length'].describe()

# Mostrar resultados
unique_speakers, duplicated_dialogs, length_stats

(<bound method Series.unique of 0         amlo
 1         amlo
 2         amlo
 3         amlo
 4         amlo
           ... 
 155502    amlo
 155503    amlo
 155504    amlo
 155505    amlo
 155506    amlo
 Name: speaker, Length: 155507, dtype: object>,
 7446,
 count    155507.000000
 mean        232.972509
 std         153.822345
 min           2.000000
 25%         112.000000
 50%         222.000000
 75%         329.000000
 max        1208.000000
 Name: length, dtype: float64)

## Segun al anilisis y En GENERAL LO QUE SE IDENTIFICO DE ESTE DATA SET FUE LO SIGUIENTE:
### Revisión de valores en la columna 'speaker'
### Se identificaron solo dos valores únicos: 'amlo' y 'lopez-gatell', lo cual es consistente con el dataset.
### Se encontraron diálogos duplicados
### Se detectaron 7,446 diálogos repetidos dentro del dataset.
### Se observó que algunos diálogos tienen solo 2 caracteres, lo que puede ser ruido o errores de transcripción.